File: flutter_app/lib/utils/plant.dart
// class Plant {
//   final int id;
//   final String scname;
//   final String cname;
//   final String cat;
//   final int maxl;
//   final int minl;
//   final int maxenvhum;
//   final int minenvhum;
//   final int maxsom;
//   final int minsom;
//   final int maxtemp;
//   final int mintemp;

//   Plant({
//     required this.id,
//     required this.scname,
//     required this.cname,
//     required this.cat,
//     required this.maxl,
//     required this.minl,
//     required this.maxenvhum,
//     required this.minenvhum,
//     required this.maxsom,
//     required this.minsom,
//     required this.maxtemp,
//     required this.mintemp,
//   });

//   @override
//   String toString() {
//     return '$scname $cname ($cat)';
//   }

//   factory Plant.fromJson(Map<String, dynamic> json){
//     return Plant(
//       id: json['id'],
//       scname: json['scientificName'],
//       cname: json['common_name'],
//       cat: json['category'],
//       maxl: json['max_light'],
//       minl: json['min_light'],
//       maxenvhum: json['max_env_humid'],
//       minenvhum: json['min_env_humid'],
//       maxsom: json['max_soil_moist'],
//       minsom: json['min_soil_moist'],
//       maxtemp: json['max_temp'],
//       mintemp: json['min_temp']
//     );
//   }
// }

class Plant {
  final int id;
  final String scientificName;
  final String commonName;
  final String category;
  final int maxLight;
  final int minLight;
  final int maxEnvHumid;
  final int minEnvHumid;
  final int maxSoilMoist;
  final int minSoilMoist;
  final int maxTemp;
  final int minTemp;

  Plant({
    required this.id,
    required this.scientificName,
    required this.commonName,
    required this.category,
    required this.maxLight,
    required this.minLight,
    required this.maxEnvHumid,
    required this.minEnvHumid,
    required this.maxSoilMoist,
    required this.minSoilMoist,
    required this.maxTemp,
    required this.minTemp,
  });

  factory Plant.fromJson(Map<String, dynamic> json) {
    return Plant(
      id: json['id'],
      scientificName: json['scientificName'],
      commonName: json['common_name'],
      category: json['category'],
      maxLight: json['max_light'],
      minLight: json['min_light'],
      maxEnvHumid: json['max_env_humid'],
      minEnvHumid: json['min_env_humid'],
      maxSoilMoist: json['max_soil_moist'],
      minSoilMoist: json['min_soil_moist'],
      maxTemp: json['max_temp'],
      minTemp: json['min_temp'],
    );
  }

  @override
  String toString() {
    return '$scientificName ($commonName, $category)';
  }
}

class PlantInstance {

  final int id;
  final Plant plant;
  final String? nickname;

  PlantInstance({
    required this.id,
    required this.plant,
    this.nickname
  });

  factory PlantInstance.fromJson(Map<String, dynamic> json) {
    return PlantInstance(
      id: json['id'],
      plant: Plant.fromJson(json['plant']),
      nickname: json['nickname'],
    );
  }
}

File: szoftarch.backend/src/main/java/backend/Model/Plant.java
package backend.Model;

import com.fasterxml.jackson.annotation.JsonProperty;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "plants")
public class Plant {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    @JsonProperty
    private String  scientificName, common_name, category;
    @JsonProperty 
    private Integer max_light, min_light, max_env_humid, min_env_humid, max_soil_moist, min_soil_moist, max_temp, min_temp;

    //Getterek setterek
    public String getScientificName() {
        return scientificName;
    }

    public void setScientificName(String scientific_name) {
        this.scientificName = scientific_name;
    }

    public String getCommon_name() {
        return common_name;
    }

    public void setCommon_name(String common_name) {
        this.common_name = common_name;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }
    
    public Integer getMax_light() {
        return max_light;
    }

    public void setMax_light(Integer max_light) {
        this.max_light = max_light;
    }

    public Integer getMin_light() {
        return min_light;
    }

    public void setMin_light(Integer min_light) {
        this.min_light = min_light;
    }

    public Integer getMax_env_humid() {
        return max_env_humid;
    }

    public void setMax_env_humid(Integer max_env_humid) {
        this.max_env_humid = max_env_humid;
    }

    public Integer getMin_env_humid() {
        return min_env_humid;
    }

    public void setMin_env_humid(Integer min_env_humid) {
        this.min_env_humid = min_env_humid;
    }

    public Integer getMax_soil_moist() {
        return max_soil_moist;
    }

    public void setMax_soil_moist(Integer max_soil_moist) {
        this.max_soil_moist = max_soil_moist;
    }

    public Integer getMin_soil_moist() {
        return min_soil_moist;
    }

    public void setMin_soil_moist(Integer min_soil_moist) {
        this.min_soil_moist = min_soil_moist;
    }

    public Integer getMax_temp() {
        return max_temp;
    }

    public void setMax_temp(Integer max_temp) {
        this.max_temp = max_temp;
    }

    public Integer getMin_temp() {
        return min_temp;
    }

    public void setMin_temp(Integer min_temp) {
        this.min_temp = min_temp;
    }

    public Plant() {}

    public Plant(String scientific_name2, String common_name2, String category2, Integer max_light2, Integer min_light2,
            Integer max_env_humid2, Integer min_env_humid2, Integer max_soil_moist2, Integer min_soil_moist2,
            Integer max_temp2, Integer min_temp2) {
        this.scientificName = scientific_name2;
        this.common_name = common_name2;
        this.category = category2;
        this.max_light = max_light2;
        this.min_light = min_light2;
        this.max_env_humid = max_env_humid2;
        this.min_env_humid = min_env_humid2;
        this.max_soil_moist = max_soil_moist2;
        this.min_soil_moist = min_soil_moist2;
        this.max_temp = max_temp2;
        this.min_temp = min_temp2;

    }

}
